data_model/src/transaction.rs

```diff
@@ -259,17 +259,16 @@ declare_versioned!(SignedTransaction 1..2, Debug, Display, Clone, PartialEq, Eq,
 declare_versioned!(SignedTransaction 1..2, Debug, Display, Clone, PartialEq, Eq, PartialOrd, Ord, FromVariant, IntoSchema);
 
 impl SignedTransaction {
-    /// Return transaction payload
-    // FIXME: Leaking concrete type TransactionPayload from Versioned container. Payload should be versioned
-    pub fn payload(&self) -> &TransactionPayload {
+    /// Define the version of SignedTransaction
+    /// Created to provide access to the transaction payload only to internal libs
+    // TODO: Make a correct implementation
+    pub fn transaction(&self) -> &SignedTransactionV1 {
         let SignedTransaction::V1(tx) = self;
-        &tx.payload
+        tx
     }
```

Comment:
remove this function or make it private or guard it with `transparent_api` macro. Client must not get a handle to `SignedTransactionV1`